python/ml4ir/base/model/architectures/auto_dag_network.py

```diff
@@ -0,0 +1,355 @@
+from __future__ import annotations
+
+from typing import List, Dict, Union, Type
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from ml4ir.base.config.keys import FeatureTypeKey
+from ml4ir.base.features.feature_config import FeatureConfig
+from ml4ir.base.io.file_io import FileIO
+
+
+def get_layer_subclasses() -> Dict[str, Type[layers.Layer]]:
+    """Get mapping of {subclass-name: subclass} for all derivative classes of keras.layers.Layer"""
+
+    def full_class_name(cls):
+        """Get {package_name}.{class_name}"""
```

Comment:
@arvindsrikantan Can you add an example for keras native classes and custom ml4ir classes here?